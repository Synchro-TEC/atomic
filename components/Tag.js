import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Tag = ({ theme, className, onClose, children, ...otherProps }) => {
  const baseClass = `sv-tag ${theme}`;
  const rootClass = cx(baseClass, className);

  return (
    <div className={rootClass} {...otherProps}>
      <span className="sv-tag__close" onClick={onClose}>
        Ã—
      </span>
      <span className="sv-tag__content">{children}</span>
    </div>
  );
};

Tag.defaultProps = {
  theme: 'default',
};

Tag.propTypes = {
  onClose: PropTypes.func,
  theme: PropTypes.oneOf(['default', 'primary', 'danger', 'info', 'warning']),
};

export default Tag;
