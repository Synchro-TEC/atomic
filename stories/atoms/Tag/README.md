The `<Tag>` component is a button.