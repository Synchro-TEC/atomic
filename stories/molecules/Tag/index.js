import React from 'react';
// import { text, boolean, object } from "@storybook/addon-knobs";
import { action } from '@storybook/addon-actions';
import { optionalSelect } from '../../../utils/optionalSelect';

import readme from './README.md';
import Tag from '../../../components/Tag';

const options = {
  default: 'default',
  primary: 'primary',
  info: 'info',
  warning: 'warning',
  danger: 'danger',
};

const component = () => (
  <Tag theme={optionalSelect('Theme', options, '')} onClose={action('close clicked')}>
    tagname
  </Tag>
);

export default [readme, component];
