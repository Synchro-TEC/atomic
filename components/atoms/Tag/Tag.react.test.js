import React from 'react';
import Tag from './Tag';

describe('Tag Component', () => {
  test('Render the button', () => {
    const tag = shallow(<Tag>The Tag</Tag>);
    expect(tag).toMatchSnapshot();
  });

  test('Fire onClick button', () => {
    const mockCallBack = jest.fn();
    const tag = shallow(<Tag onClose={mockCallBack}>The Tag</Tag>);
    tag.find('.sv-tag__close').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
