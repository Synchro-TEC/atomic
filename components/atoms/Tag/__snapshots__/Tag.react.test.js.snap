// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component Render the button 1`] = `
<div
  className="sv-tag default"
>
  <span
    className="sv-tag__close"
  >
    Ã—
  </span>
  <span
    className="sv-tag__content"
  >
    The Tag
  </span>
</div>
`;
